### ISSUE TITLE
Bot selecting wrong pokèmon

### ISSUE THREAD

**Issue by HolidayOugi** [imaboy321/Ledybot#27]
The bot works great at first but after a while it begins accepting trade requests with the wrong pokèmon, resulting in the whole thing crashing.

Using Old3DS, already using the workaround explained in #24

**Comment by gr33nd3v1l** [comment]
Have you tried the stability improvement by using CheatEngine? #20 

This is usually the bot getting its timers wrong so you end up clicking the wrong thing and getting in the wrong places

**Comment by HolidayOugi** [comment]
Honestly didn't, will try this afternoon

**Comment by HolidayOugi** [comment]
Still doesn't work. Ledybot recognizes the right trainer (it appears on the list) but it always chooses the first GTS match regardless which is often the wrong one. Tried with different speeds, same result. Cheat Engine was working because the slowdown was evident at slower speeds

**Comment by HolidayOugi** [comment]
Also, it often gives me this message (half in italian but should be understandable): 
System.ObjectDisposedException: Impossibile accedere a un oggetto eliminato.
Nome oggetto: 'System.Net.Sockets.NetworkStream'.
   in System.Net.Sockets.NetworkStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   in Ledybot.NTR.sendPacket(UInt32 type, UInt32 cmd, UInt32[] args, UInt32 dataLen)
   in Ledybot.NTR.sendWriteMemPacket(UInt32 addr, UInt32 pid, Byte[] buf)
   in Ledybot.RemoteControl.<quickbuton>d__17.MoveNext()
--- Fine traccia dello stack da posizione precedente dove è stata generata l'eccezione ---
   in System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)


************** Assembly caricati **************
mscorlib
    Versione assembly: 4.0.0.0
    Versione Win32: 4.6.1637.0 built by: NETFXREL3STAGE
    Base di codice: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
Ledybot
    Versione assembly: 1.0.0.0
    Versione Win32: 1.0.0.0
    Base di codice: file:///C:/Users/emy40/Desktop/612%20%E2%98%85%20-%20Haxorus%20-%20CC21B7241142/Ledybot.exe
----------------------------------------
System.Windows.Forms
    Versione assembly: 4.0.0.0
    Versione Win32: 4.6.1586.0 built by: NETFXREL2

**Comment by gr33nd3v1l** [comment]
It doesn't _actually_ select the wrong trainer! The pointer never moves to the right trainer because the bot injects the trade so to speak.

The second error message is usually what I get when the file for the pokemon being traded doesn't exist.
Try moving the .pk7 to desktop and making sure it works.

Here's a [Docile Ditto file](https://www.dropbox.com/s/46suq0y4u6tmqsc/DITTO.pk7?dl=1) that you can trade for Ledyba that definitely works

**Comment by HolidayOugi** [comment]
The .pk7 file was already on the Desktop. The bot works fine is the trade is the first on the GTS list. However, if thats not the case, its going to initiate a trade but I won't be able to trade the pokèmon, which has been injected anyway. So really dunno

**Comment by HolidayOugi** [comment]
_Update: now it doesn't recognize any trade at all, even if it is the first of the list..._

Fixed by moving the Ledybot exe

**Comment by gr33nd3v1l** [comment]
Are you sure you're running the correct versions of the 3ds-related software? Sounds like a connection problem

**Comment by HolidayOugi** [comment]
Of course, using BootNTR Selector Mode 3 with the latest NTR version and Input Redirection Mode 3.

**Comment by HolidayOugi** [comment]
Still stuck with the same problem :/

**Comment by HolidayOugi** [comment]
Trieid today with a different setup. Nothing at all...
